Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -1734,23 +1726,7 @@ private boolean isSessionAvailableForRequest() {
         return (hasSession() && hasDeviceFingerPrint());
     }
     
-    private void handleFailure(int index, int statusCode) {
```

Comment:
bloat + see comment in BranchPostTask wrt unnecessary use of indexes (i.e. the places where this is called all have a reference to the request that needs to handle the error)